x-pack/plugins/security_solution/server/lib/detection_engine/signals/threat_mapping/create_threat_signal.ts

```diff
@@ -43,7 +43,7 @@ export const createThreatSignal = async ({
     threatList: currentThreatList,
   });
 
-  if (threatFilter.query.bool.should.length === 0) {
+  if (threatFilter.query?.bool.should.length === 0) {
```

Comment:
I think we want `threatFilter.query?.bool.should.length === 0` || threatFilter.query == null` at this point if the query and boolean aren't guaranteed to be existing.

Reply:
This shouldn't happen, but you are correct!